test/jdk/javax/swing/text/html/CSS/bug4174871.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JFrame;
+import javax.swing.JTextPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.View;
+import java.awt.Robot;
+import java.awt.Shape;
+
+/*
+ * @test
+ * @bug 4174871
+ * @key headful
+ * @summary  Tests if CELLSPACING attribute in HTML table is rendered
+ */
+
+public class bug4174871 {
+    private static JFrame frame;
+    private static JTextPane pane;
+    private static volatile boolean passed = false;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+
+            SwingUtilities.invokeAndWait(bug4174871::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(500);
+
+            SwingUtilities.invokeAndWait(bug4174871::testUI);
+            robot.waitForIdle();
+            robot.delay(200);
```

Comment:
After you've performed all the actions, waiting doesn't seem to affect anything, and therefore it could be removed.

This comment also applies to the tests below.